xonsh/main.py

```diff
@@ -252,10 +253,58 @@ def premain(argv=None):
     return args
 
 
+def _failback_to_other_shells(argv):
+    args = None
+    try:
+        args = premain(argv)
+    except Exception:
+        pass
+    # only failback for interactive shell; if we cannot tell, treat it
+    # as an interactive one for safe.
+    if hasattr(args, 'mode') and args.mode != XonshMode.interactive:
+        sys.exit(1)
```

Comment:
Since `premain()` is the first expression (from code or logic phases) in `main_xonsh()`, so if it raises errors the traceback is already printed; if not, then there should be no exceptions too.  That's my idea here, I can add reraise here is you still think the reason I just put is not valid enough :)